packages/multichain/src/scope/assert.test.ts

```diff
@@ -480,4 +481,82 @@ describe('Scope Assert', () => {
       ).toThrow('ExternalScopeObject.rpcEndpoints must be an array of strings');
     });
   });
+
+  describe('assertIsInternalScopesObject', () => {
+    it('does not throw if passed obj is a valid InternalScopesObject with all valid properties', () => {
+      const obj = {
+        'eip155:1': {
+          accounts: ['eip155:1:0x1234'],
+        },
+      };
+      expect(() => assertIsInternalScopesObject(obj)).not.toThrow();
+    });
+
+    it('throws an error if passed obj is not an object', () => {
+      expect(() => assertIsInternalScopesObject(null)).toThrow(
+        'InternalScopesObject must be an object',
+      );
+      expect(() => assertIsInternalScopesObject(123)).toThrow(
+        'InternalScopesObject must be an object',
+      );
+      expect(() => assertIsInternalScopesObject('string')).toThrow(
+        'InternalScopesObject must be an object',
+      );
+    });
+
+    it('throws and error if passed an object with an InternalScopeObject value that is not an object', () => {
+      expect(() => assertIsInternalScopesObject({ 'eip155:1': 123 })).toThrow(
+        'InternalScopeObject must be an object',
+      );
+    });
+
+    it('throws an error if passed an object with a key that is not a valid InternalScopeString', () => {
+      jest.spyOn(Utils, 'isCaipReference').mockImplementation(() => false);
+
+      expect(() =>
+        assertIsInternalScopesObject({ 'invalid-scope-string': {} }),
+      ).toThrow('scopeString is not a valid InternalScopeString');
+    });
+
+    it('throws an error if passed an object with an InternalScopeObject without an accounts property', () => {
+      const invalidInternalScopeObject = {
+        'eip155:1': {},
+      };
+      expect(() =>
+        assertIsInternalScopesObject(invalidInternalScopeObject),
+      ).toThrow(
+        'InternalScopeObject.accounts must be an array of CaipAccountId',
+      );
+    });
+
+    it('throws an error if passed an object with an InternalScopeObject with an accounts property that is not an array', () => {
+      const invalidInternalScopeObject = {
+        'eip155:1': {
+          accounts: 'not-an-array',
+        },
+      };
+      expect(() =>
+        assertIsInternalScopesObject(invalidInternalScopeObject),
+      ).toThrow(
+        'InternalScopeObject.accounts must be an array of CaipAccountId',
+      );
+    });
+
+    it('throws an error if accounts contains invalid CaipAccountId', () => {
+      const invalidInternalScopeObject = {
+        'eip155:1': {
+          accounts: ['eip155:1:0x1234', 'invalidAccount'],
+        },
+      };
+      MockUtils.isCaipAccountId.mockImplementation(
+        (id) => id !== 'invalidAccount',
+      );
+      expect(() =>
+        assertIsInternalScopesObject(invalidInternalScopeObject),
+      ).toThrow(
+        'InternalScopeObject.accounts must be an array of CaipAccountId',
+      );
+      jest.restoreAllMocks();
```

Comment:
[nit] would it make more sense to throw this in a afterEach() hook?

Reply:
done here https://github.com/MetaMask/core/pull/4950/commits/15004433284affd28973e3c6bac2f8361eb70766